apps/hash-frontend/src/shared/layout/layout-with-sidebar/account-entity-type-list/search-input.tsx

```diff
@@ -0,0 +1,98 @@
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+import {
+  FontAwesomeIcon,
+  IconButton,
+  TextField,
+} from "@hashintel/design-system";
+import { Fade, outlinedInputClasses, Tooltip } from "@mui/material";
+import { FunctionComponent } from "react";
+
+import { MagnifyingGlassLightIcon } from "../../../icons/magnifying-glass-light";
+
+type SearchInputProps = {
+  searchVisible: boolean;
+  showSearchInput: () => void;
+  hideSearchInput: () => void;
+  onChangeText: (text: string) => void;
+};
+
+export const SearchInput: FunctionComponent<SearchInputProps> = ({
+  searchVisible,
+  showSearchInput,
+  hideSearchInput,
+  onChangeText,
+}) => (
+  <>
+    <Tooltip title="Search for types">
+      <IconButton
+        size="medium"
+        sx={({ palette }) => ({
+          color: palette.gray[50],
+          svg: { fontSize: 14 },
+        })}
+        onClick={() => showSearchInput()}
+      >
+        <MagnifyingGlassLightIcon />
+      </IconButton>
+    </Tooltip>
+    <Fade in={searchVisible}>
+      <TextField
```

Comment:
Can we add autofocus of this? It wasn't autofocusing before because it was omnipresent, but now that it appears on button click I think it should autofocus

Reply:
Good idea, have addressed in https://github.com/hashintel/hash/pull/2908/commits/d99e2819e61f3c6cf0c5ed09461ef4f5b6068cd2